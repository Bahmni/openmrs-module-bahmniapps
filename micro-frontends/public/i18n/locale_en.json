{
  "NO_FORM": "No Form found for this patient....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Observation Forms",
  "LOADING_MESSAGE": "Loading... Please Wait",
  "NO_ALLERGIES": "No Allergies recorded for this patient.",
  "BACK_TO_ALLERGEN": "Back to Allergies",
  "ALLERGIES_HEADING": "Allergies and Reactions",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergies",
  "SEVERITY": "Severity",
  "SAVE": "Save",
  "CANCEL": "Cancel",
  "NO_ALLERGENS_FOUND": "No Allergen found",
  "SEARCH_ALLERGEN": "Search Allergen",
  "REACTIONS": "Reaction(s)",
  "SEARCH_REACTION": "Search Reaction",
  "COMMON_REACTIONS": "Common Reactions",
  "ALLERGEN": "Allergen",
  "EDIT_FORM_ERROR_MESSAGE": "Please enter a value in the mandatory fields or correct the value in the highlighted fields to proceed",
  "EMPTY_NOTES_ERROR": "Note cannot be empty",
  "OT_NOTES": "Notes",
  "FROM": "From",
  "TO": "To",
  "FROM_DATE_BEFORE_TO_DATE_ERROR": "From date should be before To date",
  "DATE_OUT_OF_RANGE_ERROR": "Please select date within the valid range",
  "DELETE_NOTE_TITLE": "Delete Note",
  "YES_KEY": "Yes",
  "NO_KEY": "No",
  "DELETE_NOTE_MESSAGE": "Are you sure you want to delete this note?",
  "NOTES_PLACEHOLDER": "Enter a maximum of 150 characters",
  "UPDATE_NOTE_TITLE": "Update Note",
  "ADD_NOTE_TITLE": "Add Note",
  "LOADING": "Loading...",
  "UPDATE": "Update",
  "AKNOWLEDGE_BUTTON": "Acknowledge",
  "AKNOWLEDGEMENT_REQUIRED_TEXT": "Acknowledgement required",
  "NO_DRUGS_TO_BE_AKNOWLEDGED_TEXT": "You have no new drugs to be acknowledged"
}