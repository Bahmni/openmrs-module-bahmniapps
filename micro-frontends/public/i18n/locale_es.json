{
  "NO_FORM": "No se encontró ningún formulario para este paciente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formularios de observación",
  "LOADING_MESSAGE": "Cargando... Espere por favor",
  "NO_ALLERGIES": "No se registraron alergias para este paciente.",
  "BACK_TO_ALLERGEN": "Volver a Alergias",
  "ALLERGIES_HEADING": "Alergias y reacciones",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Alergias",
  "SEVERITY": "Gravedad",
  "SAVE": "Ahorrar",
  "CANCEL": "Cancelar",
  "NO_ALLERGENS_FOUND": "No se encontró ningún alérgeno",
  "SEARCH_ALLERGEN": "Buscar alérgeno",
  "REACTIONS": "Reacciones",
  "SEARCH_REACTION": "Reacción de búsqueda",
  "COMMON_REACTIONS": "Reacciones comunes",
  "ALLERGEN": "Alérgeno",
  "EDIT_FORM_ERROR_MESSAGE": "Ingrese un valor en los campos obligatorios o corrija el valor en los campos resaltados para continuar"
}
