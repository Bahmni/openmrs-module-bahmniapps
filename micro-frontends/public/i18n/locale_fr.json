{
  "NO_FORM": "Aucun formulaire trouvé pour ce patient....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formulaires d'observations",
  "LOADING_MESSAGE": "Chargement... Veuillez patienter",
  "NO_ALLERGIES": "Aucune allergie enregistrée pour ce patient.",
  "BACK_TO_ALLERGEN": "Retour à Allergies",
  "ALLERGIES_HEADING": "Allergies et réactions",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergies",
  "SEVERITY": "Gravité",
  "SAVE": "Sauvegarder",
  "CANCEL": "Annuler",
  "NO_ALLERGENS_FOUND": "Aucun allergène trouvé",
  "SEARCH_ALLERGEN": "Rechercher un allergène",
  "REACTIONS": "Réaction(s)",
  "SEARCH_REACTION": "Réaction à la recherche",
  "COMMON_REACTIONS": "Réactions courantes",
  "ALLERGEN": "Allergène",
  "EDIT_FORM_ERROR_MESSAGE": "Veuillez saisir une valeur dans les champs obligatoires ou corriger la valeur dans les champs en surbrillance pour continuer",
  "ACKNOWLEDGEMENT": "Reconnaître",
  "ENTER_NOTES": "Saisir des notes",
  "NOTE": "Note",
  "ALLERGY_SAVED_SUCCESS": "Allergie enregistrée avec succès",
  "ERROR_SAVING_ALLERGY": "Erreur lors de l'enregistrement de l'allergie",
  "ACKNOWLEDGEMENT_REQUIRED_TEXT": "Accusé de réception requis",
  "MALE": "Homme",
  "FEMALE": "Femme",
  "OTHER": "Autre",
  "ADDITIONAL_COMMENT_ALLERGY": "Commentaires supplémentaires tels que la date d'apparition, etc.",
  "ALLERGEN_SEARCH_PLACEHOLDER": "Saisissez pour rechercher un allergène"
}
