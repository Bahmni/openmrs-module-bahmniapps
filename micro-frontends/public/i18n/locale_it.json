{
  "NO_FORM": "Nessun modulo trovato per questo paziente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Moduli di osservazione",
  "LOADING_MESSAGE": "Caricamento in corso... Attendere prego",
  "NO_ALLERGIES": "Nessuna allergia registrata per questo paziente.",
  "BACK_TO_ALLERGEN": "Torniamo alle allergie",
  "ALLERGIES_HEADING": "Allergie e reazioni",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Allergie",
  "SEVERITY": "Gravit√†",
  "SAVE": "Salva",
  "CANCEL": "Cancellare",
  "NO_ALLERGENS_FOUND": "Nessun allergene trovato",
  "SEARCH_ALLERGEN": "Cerca allergene",
  "REACTIONS": "Reazione(i)",
  "SEARCH_REACTION": "Cerca reazione",
  "COMMON_REACTIONS": "Reazioni comuni",
  "ALLERGEN": "Allergene",
  "EDIT_FORM_ERROR_MESSAGE": "Inserisci un valore nei campi obbligatori o correggi il valore nei campi evidenziati per procedere"
}
