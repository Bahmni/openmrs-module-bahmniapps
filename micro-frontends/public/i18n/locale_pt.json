{
  "NO_FORM": "Nenhum formulário encontrado para este paciente....",
  "DASHBOARD_TITLE_FORMS_2_DISPLAY_CONTROL_KEY": "Formulários de Observação",
  "LOADING_MESSAGE": "Carregando... Aguarde",
  "NO_ALLERGIES": "Nenhuma alergia registrada para este paciente.",
  "BACK_TO_ALLERGEN": "Voltar para Alergias",
  "ALLERGIES_HEADING": "Alergias e reações",
  "ALLERGIES_DISPLAY_CONTROL_HEADING": "Alergias",
  "SEVERITY": "Gravidade",
  "SAVE": "Salvar",
  "CANCEL": "Cancelar",
  "NO_ALLERGENS_FOUND": "Nenhum alérgeno encontrado",
  "SEARCH_ALLERGEN": "Pesquisar alérgeno",
  "REACTIONS": "Reação(ões)",
  "SEARCH_REACTION": "Reação de pesquisa",
  "COMMON_REACTIONS": "Reações Comuns",
  "ALLERGEN": "Alérgeno",
  "EDIT_FORM_ERROR_MESSAGE": "Insira um valor nos campos obrigatórios ou corrija o valor nos campos destacados para prosseguir"
}
