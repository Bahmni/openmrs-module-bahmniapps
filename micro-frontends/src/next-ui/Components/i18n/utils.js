import { LS_LANG_KEY, BASE_URL } from "../../constants";

const translationsBaseUrl = "i18n";

export function getLocale() {
  return localStorage.getItem(LS_LANG_KEY) || "en";;
}

export const getTranslations = async (locale) => {
  const fileName = `locale_${locale}.json`;
  return fetchTranslations(fileName);
};

async function fetchTranslations(fileName) {
  const url = `${BASE_URL}${translationsBaseUrl}/${fileName}`;
  const response = await fetch(url);
  return response.json();
}
