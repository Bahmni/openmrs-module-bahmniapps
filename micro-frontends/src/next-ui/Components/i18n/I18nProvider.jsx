import PropTypes from "prop-types";
import React, { useEffect, useMemo, useState } from "react";
import { IntlProvider } from "react-intl";
import { getLocale, getTranslations } from "./utils";

export function I18nProvider({ children }) {
  const [messages, setMessages] = useState(undefined);
  const locale = useMemo(getLocale, []);

  useEffect(() => {
    getTranslations(locale).then(setMessages);
  }, []);

  if (!messages) {
    return <div></div>;
  }

  return (
    <IntlProvider defaultLocale="en" locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
}

I18nProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
