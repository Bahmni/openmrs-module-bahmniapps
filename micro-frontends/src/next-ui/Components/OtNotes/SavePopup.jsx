import React, {useEffect, useState} from "react";
import PropTypes from "prop-types";
import {Modal, TextArea, DatePicker, DatePickerInput} from "carbon-components-react";
import "../../../styles/carbon-conflict-fixes.scss";
import "../../../styles/carbon-theme.scss";
import "./OtNotes.scss";
import {FormattedMessage, useIntl} from "react-intl";
import {I18nProvider} from '../i18n/I18nProvider';
import moment from "moment";
import {saveNote, updateNoteForADay} from "./utils";

export function SavePopup(props) {
    const {hostData, hostApi} = props;
    const {
        notes,
        weekEndDateTime = moment().endOf('day'),
        weekStartDateTime = moment().startOf('day'),
        isDayView,
        noteId,
        noteDate,
        providerUuid
    } = hostData;
    const [modalNotes, setModalNotes] = useState(notes || "");
    const [startDate, setStartDate] = useState(new Date(noteDate));
    const [endDate, setEndDate] = useState(new Date(noteDate));
    const [shouldShowErrors, setShouldShowErrors] = useState(false);
    const validMinStartDate = noteId || isDayView ? new Date(noteDate) : new Date(weekStartDateTime);
    const [validMinEndDate, setValidMinEndDate] = useState(noteId || isDayView ? new Date(noteDate) : new Date(weekEndDateTime));
    const [isLoading, setIsLoading] = useState(false);
    const intl = useIntl();

    const hasActiveErrors = () => {
        return !modalNotes || !startDate || !endDate || startDate > endDate;
    }

    const handleSave = () => {
        if (noteId) {
            updateNoteForADay(noteId, modalNotes, providerUuid).then(() => {
                setIsLoading(false);
                hostApi?.onSuccess();
            });
        } else if (isDayView) {
            saveNote(modalNotes, startDate).then(() => {
                setIsLoading(false);
                hostApi?.onSuccess();
            });
        } else {
            saveNote(modalNotes, startDate, endDate).then(() => {
                setIsLoading(false);
                hostApi?.onSuccess();
            });
        }
    }
    useEffect(() => {
        setValidMinEndDate(startDate);
    }, [startDate]);
    return (
        <I18nProvider>
            <Modal
                open
                className={"next-ui ot-notes-popup"}
                modalHeading={noteId ? <FormattedMessage id={'UPDATE_NOTE_TITLE'} defaultMessage={"Update Notes"}/> :
                    <FormattedMessage id={'ADD_NOTE_TITLE'} defaultMessage={"Add Notes"}/>}
                primaryButtonText={isLoading ?
                    <FormattedMessage id={"LOADING"} defaultMessage={"Loading..."}/> : noteId ?
                        <FormattedMessage id={"UPDATE"} defaultMessage={"Update"}/> :
                        <FormattedMessage id={"SAVE"} defaultMessage={"Save"}/>}
                secondaryButtonText={<FormattedMessage id={"CANCEL"} defaultMessage={"Cancel"}/>}
                onRequestClose={hostApi?.onClose}
                onRequestSubmit={() => {
                    setShouldShowErrors(true);
                    if (!hasActiveErrors()) {
                        setIsLoading(true);
                        handleSave();
                    }
                }}
            >
                <TextArea labelText={<FormattedMessage id={"OT_NOTES"} defaultMessage={"Notes"}/>}
                          value={modalNotes}
                          placeholder={intl.formatMessage({
                              id: "NOTES_PLACEHOLDER",
                              defaultMessage: "Enter a maximum of 150 characters"
                          })} maxCount={150} onChange={e => {
                    setModalNotes(e?.target?.value);
                }}/>
                {shouldShowErrors && !modalNotes ?
                    <p className={"error-text"}><FormattedMessage id={"EMPTY_NOTES_ERROR"}
                                                                  defaultMessage={"Note cannot be empty"}/>
                    </p> : <div style={{height: "20px"}}></div>}
                <div className={"date-range"}>
                    <div>
                        <DatePicker datePickerType="single" maxDate={new Date(weekEndDateTime)} dateFormat={"d/m/Y"}
                                    minDate={validMinStartDate} value={startDate}
                                    onChange={(e) => {
                                        if (e.length === 1) {
                                            setStartDate(e[0]);
                                        }
                                    }}>
                            <DatePickerInput
                                id="date-picker-input-id-start"
                                placeholder="dd/mm/yyyy"
                                labelText="Start date"
                                disabled={Boolean(isDayView || noteId)}
                            />
                        </DatePicker>
                        {shouldShowErrors && !startDate ?
                            <div className={"error-text"}><FormattedMessage id={'DATE_OUT_OF_RANGE_ERROR'}
                                                                            defaultMessage={"Please select date within the valid range"}/>
                            </div> : startDate > endDate ?
                                <div className={"error-text"}><FormattedMessage id={'FROM_DATE_BEFORE_TO_DATE_ERROR'}
                                                                                defaultMessage={"From date should be before To date"}/>
                                </div> :
                                <div style={{height: "20px"}}></div>
                        }
                    </div>
                    <div>
                        <DatePicker datePickerType="single" maxDate={new Date(weekEndDateTime)} dateFormat={"d/m/Y"}
                                    minDate={validMinEndDate} value={endDate}
                                    onChange={(e) => {
                                        if (e.length === 1) {
                                            setEndDate(e[0]);
                                        }
                                    }}>
                            <DatePickerInput
                                id="date-picker-input-id-finish"
                                placeholder="dd/mm/yyyy"
                                labelText="End date"
                                disabled={Boolean(isDayView || noteId)}
                            />
                        </DatePicker>
                        {shouldShowErrors && !endDate ?
                            <div className={"error-text"}><FormattedMessage id={'DATE_OUT_OF_RANGE_ERROR'}
                                                                            defaultMessage={"Please select date within the valid range"}/>
                            </div> : <div style={{height: "20px"}}></div>}
                    </div>
                </div>
            </Modal>
        </I18nProvider>
    );
}

SavePopup.propTypes = {
    hostData: PropTypes.any,
    hostApi: PropTypes.any,
}