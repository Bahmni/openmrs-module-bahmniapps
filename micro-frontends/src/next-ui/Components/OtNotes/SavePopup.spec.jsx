import React from "react";
import {fireEvent, render, screen, waitFor} from "@testing-library/react";
import {SavePopup} from "./SavePopup";
import {I18nProvider} from "../i18n/I18nProvider";

const mockSaveNote = jest.fn();
const mockUpdateNoteForADay = jest.fn();

jest.mock("../i18n/utils");
jest.mock("./utils", () => ({
    saveNote: () => mockSaveNote(),
    updateNoteForADay: () => mockUpdateNoteForADay(),
}));

describe("SavePopup", () => {
    it("should render with Add Note when noteId is empty", async () => {
        const {container, getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: '',
                    isDayView: false,
                    noteId: undefined,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        await waitFor(() => {
            expect(getByText("Add Note")).toBeTruthy();
            expect(container).toMatchSnapshot();
        });
    });
    it("should render with Update Note when noteId is defined", async () => {
        const {container, getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: 'notes for the day',
                    isDayView: false,
                    noteId: 10,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        await waitFor(() => {
            expect(getByText("Update Note")).toBeTruthy();
            expect(container).toMatchSnapshot();
        });
    });

    it("should show error when trying to save with note as empty", async () => {
        const {getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: '',
                    isDayView: false,
                    noteId: undefined,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        await waitFor(() => {
            screen.getByText("Add Note").click();
        });
        expect(() => screen.getByText("Note cannot be empty")).toThrow();
        const saveButton = getByText("Save");
        expect(saveButton).toBeTruthy();
        saveButton.click();
        expect(screen.getByText("Note cannot be empty")).toBeTruthy();
    });
    it("should call save when all data is valid", async () => {
        const {container, getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: '',
                    isDayView: false,
                    noteId: undefined,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        mockSaveNote.mockResolvedValue(() => {})
        await waitFor(() => {
            screen.getByText("Add Note").click();
        });
        const textBox = container.querySelector('.bx--text-area');
        fireEvent.change(textBox, {target: {value: 'notes for the day'}});
        const saveButton = getByText("Save");
        saveButton.click();
        await waitFor(() => {
            expect(mockSaveNote).toHaveBeenCalled();
        });
    });

    it("should call save when all data is valid in Day View", async () => {
        const {container, getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: '',
                    isDayView: true,
                    noteId: undefined,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        mockSaveNote.mockResolvedValue(() => {})
        await waitFor(() => {
            screen.getByText("Add Note").click();
        });
        const textBox = container.querySelector('.bx--text-area');
        fireEvent.change(textBox, {target: {value: 'notes for the day'}});
        const saveButton = getByText("Save");
        saveButton.click();
        await waitFor(() => {
            expect(mockSaveNote).toHaveBeenCalled();
        });
    });

    it("should call edit when noteId is present", async () => {
        const {container, getByText} = render(
            <I18nProvider>
                <SavePopup hostData={{
                    notes: 'notes for the day',
                    isDayView: false,
                    noteId: 10,
                    noteDate: new Date()
                }}/>
            </I18nProvider>
        );
        mockUpdateNoteForADay.mockResolvedValue(() => {})
        await waitFor(() => {
            screen.getByText("Update Note").click();
        });
        const textBox = container.querySelector('.bx--text-area');
        fireEvent.change(textBox, {target: {value: 'notes for the day'}});
        const saveButton = getByText("Update");
        saveButton.click();
        await waitFor(() => {
            expect(mockUpdateNoteForADay).toHaveBeenCalled();
        });
    });
});