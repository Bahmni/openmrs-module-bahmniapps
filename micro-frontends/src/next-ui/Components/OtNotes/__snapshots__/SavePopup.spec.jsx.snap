// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavePopup should render with Add Note when noteId is empty 1`] = `
<div>
  <div
    class="bx--modal bx--modal-tall is-visible next-ui ot-notes-popup"
    role="presentation"
  >
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
    <div
      aria-label="[object Object]"
      aria-modal="true"
      class="bx--modal-container"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="bx--modal-header"
      >
        
        <h3
          class="bx--modal-header__heading"
          id="bx--modal-header__heading--modal-1"
        >
          Add Note
        </h3>
        <button
          aria-label="close"
          class="bx--modal-close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--modal-close__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-labelledby="bx--modal-header__heading--modal-1"
        class="bx--modal-content"
        id="bx--modal-body--modal-1"
      >
        <div
          class="bx--form-item"
        >
          <div
            class="bx--text-area__label-wrapper"
          >
            <label
              class="bx--label"
            >
              Notes
            </label>
          </div>
          <div
            class="bx--text-area__wrapper"
          >
            <textarea
              class="bx--text-area"
              cols="50"
              placeholder="Enter a maximum of 150 characters"
              rows="4"
            />
          </div>
        </div>
        <div
          style="height: 20px;"
        />
        <div
          class="date-range"
        >
          <div>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--date-picker bx--date-picker--single"
              >
                <div
                  class="bx--date-picker-container"
                >
                  <label
                    class="bx--label"
                    for="date-picker-input-id-start"
                  >
                    Start date
                  </label>
                  <div
                    class="bx--date-picker-input__wrapper"
                  >
                    <input
                      class="bx--date-picker__input flatpickr-input"
                      id="date-picker-input-id-start"
                      pattern="\\d{1,2}\\/\\d{1,2}\\/\\d{4}"
                      placeholder="dd/mm/yyyy"
                      type="text"
                    />
                    <svg
                      aria-hidden="true"
                      class="bx--date-picker__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="height: 20px;"
            />
          </div>
          <div>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--date-picker bx--date-picker--single"
              >
                <div
                  class="bx--date-picker-container"
                >
                  <label
                    class="bx--label"
                    for="date-picker-input-id-finish"
                  >
                    End date
                  </label>
                  <div
                    class="bx--date-picker-input__wrapper"
                  >
                    <input
                      class="bx--date-picker__input flatpickr-input"
                      id="date-picker-input-id-finish"
                      pattern="\\d{1,2}\\/\\d{1,2}\\/\\d{4}"
                      placeholder="dd/mm/yyyy"
                      type="text"
                    />
                    <svg
                      aria-hidden="true"
                      class="bx--date-picker__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="height: 20px;"
            />
          </div>
        </div>
      </div>
      <div
        class="bx--modal-footer bx--btn-set"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--primary"
          tabindex="0"
          type="button"
        >
          Save
        </button>
      </div>
    </div>
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
  </div>
</div>
`;

exports[`SavePopup should render with Update Note when noteId is defined 1`] = `
<div>
  <div
    class="bx--modal bx--modal-tall is-visible next-ui ot-notes-popup"
    role="presentation"
  >
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
    <div
      aria-label="[object Object]"
      aria-modal="true"
      class="bx--modal-container"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="bx--modal-header"
      >
        
        <h3
          class="bx--modal-header__heading"
          id="bx--modal-header__heading--modal-2"
        >
          Update Note
        </h3>
        <button
          aria-label="close"
          class="bx--modal-close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--modal-close__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
      </div>
      <div
        aria-labelledby="bx--modal-header__heading--modal-2"
        class="bx--modal-content"
        id="bx--modal-body--modal-2"
      >
        <div
          class="bx--form-item"
        >
          <div
            class="bx--text-area__label-wrapper"
          >
            <label
              class="bx--label"
            >
              Notes
            </label>
          </div>
          <div
            class="bx--text-area__wrapper"
          >
            <textarea
              class="bx--text-area"
              cols="50"
              placeholder="Enter a maximum of 150 characters"
              rows="4"
            >
              notes for the day
            </textarea>
          </div>
        </div>
        <div
          style="height: 20px;"
        />
        <div
          class="date-range"
        >
          <div>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--date-picker bx--date-picker--single"
              >
                <div
                  class="bx--date-picker-container"
                >
                  <label
                    class="bx--label bx--label--disabled"
                    for="date-picker-input-id-start"
                  >
                    Start date
                  </label>
                  <div
                    class="bx--date-picker-input__wrapper"
                  >
                    <input
                      class="bx--date-picker__input flatpickr-input"
                      disabled=""
                      id="date-picker-input-id-start"
                      pattern="\\d{1,2}\\/\\d{1,2}\\/\\d{4}"
                      placeholder="dd/mm/yyyy"
                      type="text"
                    />
                    <svg
                      aria-hidden="true"
                      class="bx--date-picker__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="height: 20px;"
            />
          </div>
          <div>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--date-picker bx--date-picker--single"
              >
                <div
                  class="bx--date-picker-container"
                >
                  <label
                    class="bx--label bx--label--disabled"
                    for="date-picker-input-id-finish"
                  >
                    End date
                  </label>
                  <div
                    class="bx--date-picker-input__wrapper"
                  >
                    <input
                      class="bx--date-picker__input flatpickr-input"
                      disabled=""
                      id="date-picker-input-id-finish"
                      pattern="\\d{1,2}\\/\\d{1,2}\\/\\d{4}"
                      placeholder="dd/mm/yyyy"
                      type="text"
                    />
                    <svg
                      aria-hidden="true"
                      class="bx--date-picker__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26,4h-4V2h-2v2h-8V2h-2v2H6C4.9,4,4,4.9,4,6v20c0,1.1,0.9,2,2,2h20c1.1,0,2-0.9,2-2V6C28,4.9,27.1,4,26,4z M26,26H6V12h20	V26z M26,10H6V6h4v2h2V6h8v2h2V6h4V10z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              style="height: 20px;"
            />
          </div>
        </div>
      </div>
      <div
        class="bx--modal-footer bx--btn-set"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--primary"
          tabindex="0"
          type="button"
        >
          Update
        </button>
      </div>
    </div>
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
  </div>
</div>
`;
